Redesigned manager login

Mark-H opened the pull request:
### What does it do?
This implements a new manager login screen. Designed by Rinze van der Brug, front-end by @christianseel, back-end/js by yours truly.

![schermafbeelding 2018-02-09 om 23 49 24](https://user-images.githubusercontent.com/312944/36053883-e1ec55fa-0df3-11e8-897e-d408e1936345.jpg)

The background and logo images can be changed with settings, and some new lexicons can be used to change the help text. 

### Why is it needed?

Modernize the manager login. 

### Related issue(s)/PR(s)

N/a

Mark-H committed:
Strip down login.tpl and make it html5

christianseel commented:
The default background image was taken by myself in Engelberg, Switzerland during the MODX SnowUp. License is hereby granted for using it at the manager login page.

Mark-H commented:
Just doing some quick accessibility testing using VoiceOver, and notice that we're hiding the forms with only a height declaration, causing the form to still be accessible to screenreaders. That situation is well explained at https://cbateman.com/blog/accessible-css-transitions/ and should be solvable by adding `visibility: none;` plus a (delayed) transition on visibility.

digitalpenguin commented:
Looks great! I'm not so sure about the "feel free to login anytime" sentence though..

gpsietzema commented:
Loving it :-) I know what you mean @digitalpenguin - any ideas as an alternative?

We should also approach this as a bigger thing though. I browsed a lot of (old) issues yesterday and a lot of them mentioned lexicons/language. This is a very nice start to making MODX a little friendlier though.

digitalpenguin commented:
Not as yet unfortunately. I've been thinking about it though - it's a tough one! ;)

sottwell commented:
Anything that makes the login page easier to customize will be an improvement. Modifying the login page shouldn't require a course in ExtJS and Smarty!

christianseel commented:
Should we replace
> Remember me
with 
> Keep me logged in for 14 days
? (depending on the current lifetime settings)

christianseel commented:
It's also fully responsive…
<img width="376" alt="image 2018-02-10 at 3 33 11 pm" src="https://user-images.githubusercontent.com/482184/36063028-ca92ee30-0e77-11e8-8471-0c9328e0c91c.png">

rthrash commented:
Love, love, love this. Great work everyone—definitely more friendly!

Perhaps a single line of copy would work, including if it's your first time logging in along the lines of:

> Good (morning|afternoon|evening), log in to manage your site.

rthrash commented:
One other question comes to mind, would it be straightforward to add 2FA to this? Most implementations take you to another screen with just a 2FA code box and a checkbox to remember this device for XX days.

(A\wesome photo @christianseel, too! 🥇)

christianseel commented:
Now we also got the language switch back! Now with full native language strings instead of just lang codes.
<img width="1528" alt="bildschirmfoto 2018-02-10 um 18 04 33" src="https://user-images.githubusercontent.com/482184/36064558-11d7e81c-0e8d-11e8-8031-9eadd0644f1a.png">

rthrash commented on core/lexicon/en/login.inc.php:
```diff
@@ -38,7 +38,7 @@
 $_lang['login_greeting_evening'] = '<strong>Good evening</strong>, welcome back!';
 $_lang['login_greeting_night'] = '<strong>Good night</strong>, welcome back!';
-$_lang['login_note'] = 'Feel free to login anytime.';
+$_lang['login_note'] = 'Please login to access the manager.';
```
“Manager“ should be capitalized, and I believe it should be "log in" also.

christianseel replied:
we did a vote on "login" vs "log in" and decided both work. I'm fine with both.

christianseel replied:
`Manager` fixed. Ben, @Mark-H and @gpsietzema voted for `login` instead `log in`.

rthrash replied:
Grammatically, "log in" is a verb. A "login" is a noun or adjective.

After you enter your login to log in, you've logged in, not loginned. You can ask a friend to log you in, but it'd sound weird if you tried to get them to login you.

:P

christianseel replied:
Sounds ok for me. Updated!

wuuti commented:
I really like the idea to print how long the "Keep me logged in" cookie extension will be. So something like "Keep me logged in for the next 30 days on this device" would be nice. I now this is making the text longer, but that's way more transparent to the user (maybe we find a shorter text).

christianseel commented:
Just for @wuuti 
<img width="1443" alt="bildschirmfoto 2018-02-10 um 21 16 18" src="https://user-images.githubusercontent.com/482184/36066306-c4b4d430-0ea7-11e8-83b2-7cd494235a1d.png">
(this recognizes the `session_cookie_lifetime` setting and outputs the lifetime accordingly)

jaygilmore commented:
I absolutely love the overall treatment. 

It seems to me that a ghost button for the Forgot your Login is a) too close for to the login button, b) contains the word login and c) looks like a button. This can make it easier for an accidental click target because it's the last button in the view, and for people in a hurry, may just click whichever button. Convention generally has the Forgot Password/Login functionality triggered by a link. 

Here are a few examples: 
![login mailchimp - email marketing made easy 2018-02-12 12-57-52](https://user-images.githubusercontent.com/378199/36108960-652355c0-0ff4-11e8-85f2-ed4c9b62030b.png)
![sign in to github github - private browsing 2018-02-12 12-55-24](https://user-images.githubusercontent.com/378199/36108965-69ae5f04-0ff4-11e8-8ae3-8f39d18816e1.png)
![dribbble - sign in - private browsing 2018-02-12 12-58-59](https://user-images.githubusercontent.com/378199/36109012-8b5f426c-0ff4-11e8-88da-c91566643ac8.png)

jaygilmore commented:
Where to the help and language things go when scaled down for small viewports/mobile?
Also, what kind of element is the Language selector? It looks like a button. Should it be a select? Should it look like a select?

christianseel commented:
Language: it’s a select and currently lacks a small arrow to indicate the drop down. Just didn’t have time. Help is a button.

christianseel commented:
Screenshots after latest updates:
![bildschirmfoto 2018-02-13 um 10 08 29](https://user-images.githubusercontent.com/482184/36142868-9af91594-10a9-11e8-8e34-a92c0197fb13.png)
![image 2018-02-13 at 10 21 48 am](https://user-images.githubusercontent.com/482184/36142880-a19bc2c0-10a9-11e8-964e-ead626d57463.png)

jonleverrier commented:
Great work - loving the new manager...

I can't help but feel the help and language select fields feel a little disjointed and out of context on desktop compared to mobile. The buttons on desktop look like they do something to the large background area.

gpsietzema commented:
@jonleverrier Any suggestions?

christianseel commented:
…we could move into the white area next to the logo? Through I find personally that this makes them less discoverable.

Jako commented:
I prefer the same location as on the mobile screen (top of the modx logo).

jonleverrier commented:
@gpsietzema I think the mobile version has the right hierarchy of information (what @Jako said). 

Great work guys!

rthrash commented:
I'm a fan of the mobile layout as well for the buttons/language selector on the desktop also.

Mark-H commented:
I'd say this is ready to get merged in. Per earlier discussions (primarily in Slack/in person at the SnowUp) this should be going into 3.x instead of 2.x, so it's part of the bigger MODX3 effort happening right now. 

Regarding things like the button positions, we can continue to make changes in separate PRs. It's a pretty massive improvement, so it's good that there's discussion among design-minded folks about the fine details, so let's continue that in an issue or other medium and iterate on what's there.

Mark-H committed:
Implement new login design for the manager

Based on a design by Rinze van der Brug, implemented by Christian Seel and Mark Hamstra

Squashed commit of the following:

commit 86ca599bbea1996114c48204e9be17a473c5c9cd
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 16 20:53:01 2018 +0100

    Add missing site url to "Return to website" link

commit 81b47ae3c3bd112160ff47f562d991da45e3bb54
Author: Christian Seel <cs@seda.digital>
Date:   Tue Feb 13 10:41:37 2018 +0100

    add missing c-form classes

commit ef57983e0181f018b45d732fdbc6272ae720f4a1
Author: Christian Seel <cs@seda.digital>
Date:   Tue Feb 13 10:33:58 2018 +0100

    make styles work without help button

commit e6c26aed3d1792b99e08f5aa0c69fa8b8f626ace
Merge: bc3e1b06e 15c713c42
Author: Christian Seel <cs@seda.digital>
Date:   Tue Feb 13 10:26:41 2018 +0100

    Merge branch 'new-login-screen' of https://github.com/Mark-H/revolution into new-login-screen

    # Conflicts:
    #	manager/templates/default/security/login.tpl

commit bc3e1b06e7ccd18e32a549b2a4ce14a9bdc3bdaf
Author: Christian Seel <cs@seda.digital>
Date:   Tue Feb 13 10:23:07 2018 +0100

    optimized language select, added return to website and optimized styles to follow design draft

commit 15c713c42d82f642c494d1761973546d45e2c6ff
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Mon Feb 12 13:23:39 2018 +0100

    Automatically reload the page after changing the language

commit 2f700ab0218df003180aa7ac3b7ef214bd1fc2fd
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Mon Feb 12 13:23:04 2018 +0100

    Make sure language selector automatically falls back on different languages

commit aaa204458b03f5e41c654cf018abb500aa3baff1
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Mon Feb 12 13:22:02 2018 +0100

    Add setting to control if the help button shows up, defaulting to off

commit 7f4e02765aa82d0f0398e465290c2a88e2807b59
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 22:18:33 2018 +0100

    remove debug tpl

commit d57ccb9b48dcad4c2f86fb65b32bbf37ec0ab57b
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 21:26:40 2018 +0100

    log in

commit 9c8664603dd8b931252cd94e52c668eb2008e857
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 21:15:47 2018 +0100

    include lifetime to rememberme string

commit 6b74f057d3ed11e99ed61e16808016e131c6df83
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 18:56:00 2018 +0100

    It’s a Manager

commit 5e347ce07569c54e7b75506bff19bfe8c39b9e66
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 18:12:46 2018 +0100

    improve wording on login screen

commit c4e69b3ad89ba17bb7b122d9987cb5c08a088a1c
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 18:00:38 2018 +0100

    make language switch response and use native language strings instead of lang codes

commit e1dbac09f8adaecd03c0c3c1f3233f9ddbcff93a
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 16:43:27 2018 +0100

    no more uppercase language codes

commit 97e6eecaa5282bdb4d528a1d8970080ae18059b1
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 16:39:36 2018 +0100

    add help and language switch

commit be18c9901b09cb92e1c96eb38ef907ed0458539b
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 15:52:54 2018 +0100

    refactor css classes to BEM

commit a067f337d5adafe4168034a8086d291096e6be61
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 14:06:16 2018 +0100

    responsive optimizations

commit 72cc9f1a6cccebf1b02993b5c693085d5e896b24
Author: Christian Seel <cs@seda.digital>
Date:   Sat Feb 10 13:21:12 2018 +0100

    set line-height for paragraphs

commit 0c292881e4048e196f40a74621214f9616b66a06
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Sat Feb 10 01:37:06 2018 +0100

    Remove extjs dependencies, and move the only required script into the footer

commit cbd6e877341eed411eba347067f916d194dd4965
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Sat Feb 10 01:09:21 2018 +0100

    Add missing lexicon and restore former IDs in the form

commit 1f05dd84e073b2b6aa06f4c846aca2d05a3e1cfc
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Sat Feb 10 00:55:31 2018 +0100

    Automatically focus on the username field in the new form when toggling

commit e1acacae77859928a3e242a0bca85709b181661c
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Sat Feb 10 00:55:08 2018 +0100

    Hide hidden elements from screenreaders/keyboard navigation for improved accessibility

commit e27c8e2450c268df2eec0b3936b5e75d40abd008
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Sat Feb 10 00:47:25 2018 +0100

    Cleanup unused CSS/JS, remove dependency on ExtJS even more

commit 878e35fdaf128b341f197677538be325cdbe68e8
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 23:47:18 2018 +0100

    Add lexicons for the help button

commit 82b1ad612b68e2d5df4593313f345ed2685a6733
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 23:36:52 2018 +0100

    typo

commit d6e14b97618529ff10e5c7bcb77fdeb3ecd68547
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 23:35:32 2018 +0100

    bring back marks placeholders

commit 30dfce25b179faf5c0aff403b6a378a13cc0fa89
Merge: 645bd5194 881338626
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 23:34:45 2018 +0100

    Merge branch 'new-login-screen' of https://github.com/Mark-H/revolution into new-login-screen

    # Conflicts:
    #	manager/templates/default/security/login.tpl

commit 645bd5194fcfbe6b74c384137b1fc3ff561e777a
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 23:34:04 2018 +0100

    added help link

commit 881338626453e36cce9b7290d0dd746d28e5b31d
Merge: 50b2f946a 81df10f9a
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 23:27:29 2018 +0100

    Merge branch 'new-login-screen' of github.com:Mark-H/revolution into new-login-screen

commit 50b2f946af3ac988a19334877afcec91a4a72150
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 23:27:22 2018 +0100

    Add login_background_image and login_logo settings to change what's shown in the manager login without needing a custom theme

commit 81df10f9a8816f315e6c434bb3202121c690671e
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 23:18:07 2018 +0100

    increase max-height for toggle

commit 70d5f4f91f409d7805fec0ab801c547998180b55
Merge: 480c59ce9 597f0a31b
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 23:16:11 2018 +0100

    Merge branch 'new-login-screen' of https://github.com/Mark-H/revolution into new-login-screen

commit 480c59ce929f42f32798e75764313a65f50a4a8a
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 23:15:59 2018 +0100

    stuff - best commit message ever

commit 597f0a31b9c056ae1bcd70240c5500101e94bdde
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 23:13:33 2018 +0100

    Make sure captcha works

commit 7b3a1096b1dde156fefb4e2308d341109a846875
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 23:06:07 2018 +0100

    Add missing hidden fields

commit 8dee01304771f4ca08a67b0849804abdc8f94139
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 23:01:37 2018 +0100

    Make sure forms show errors

commit dcd988097daa315e55d250da06c4502d6d2725e9
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:53:23 2018 +0100

    optimize

commit 7086fd18e9bacb0acab0ccf708b8ac977774b200
Merge: 692cc70a8 1c5e700d0
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:51:17 2018 +0100

    Merge branch 'new-login-screen' of https://github.com/Mark-H/revolution into new-login-screen

commit 1c5e700d05bb3506937f59728bc6e2d365988684
Merge: d0d0f5793 8e075be22
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 22:51:01 2018 +0100

    Merge remote-tracking branch 'origin/new-login-screen' into new-login-screen

commit 692cc70a8146ce40e1e3df37a519e00e34e56d57
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:50:44 2018 +0100

    optimize outline

commit d0d0f57938a31edc8d36d13f809541045919a0d4
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 22:49:52 2018 +0100

    Refactor extjs into vanillajs

commit 8e075be226e3d51919fc73bd93d30b7d9dd95262
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:45:27 2018 +0100

    bugfix form margin

commit ba45caa662d07445a84aef33bf194e22ad9e326b
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:44:00 2018 +0100

    optimize

commit 79233dcc57be05dfa437b6134733e177ec4dcf45
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:39:56 2018 +0100

    can-toggle

commit bbec2bf0f9a99b44fc884f437b0971b607213674
Merge: 81f22fcb2 c9bae60b6
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 22:33:01 2018 +0100

    Merge branch 'new-login-screen' of github.com:Mark-H/revolution into new-login-screen

commit 81f22fcb2f09a4a3f709019a107c6a5c4dc74ae8
Merge: c13cc00b2 0dbcebad0
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 22:32:47 2018 +0100

    Merge branch 'new-login-screen' of github.com:Mark-H/revolution into new-login-screen

commit c9bae60b6f2996d7c02e04c62fa414a14c9832d3
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:30:43 2018 +0100

    basic toggle transition

commit c13cc00b24a0c37b99ad0e06c193cbfbc204bbbb
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 22:30:27 2018 +0100

    Work on showing/hiding stuff properly

commit 0dbcebad06405dff758bad815e730c45062d2d9a
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:21:04 2018 +0100

    make this thing a button

commit a0ff1fb525f066cf8ab2d0f39f67535ef364a8bc
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:10:05 2018 +0100

    wip on forgot login

commit 9fb3969ab1c863392cf0e91f746b0cc5f056a871
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 22:00:06 2018 +0100

    wip login screen

commit 6dd8517800e742df6e26dabb921d59740b6f7483
Merge: e127afe35 124fec07d
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 21:46:46 2018 +0100

    Merge branch 'new-login-screen' of https://github.com/Mark-H/revolution into new-login-screen

    # Conflicts:
    #	manager/templates/default/security/login.tpl

commit e127afe3542d4c405cdcd37edef47d80d50e7c36
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 21:45:18 2018 +0100

    wip login screen

commit 124fec07da3c6f7432a16d6471ef61142af61156
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 21:44:11 2018 +0100

    Use lexicon for the welcome note as well

commit b7e401836f7f02931780c2978c3866bec752dba8
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 21:42:59 2018 +0100

    Make greeting dynamic based on server time + offset

commit 5db31885acba94d829249663c64e1e03e6d144e4
Author: Christian Seel <cs@seda.digital>
Date:   Fri Feb 9 21:31:08 2018 +0100

    wip login frontend

commit 83e18f6acb91b57957a4fa8e67a91e305d89d5e8
Author: Mark Hamstra <hello@markhamstra.com>
Date:   Fri Feb 9 20:37:53 2018 +0100

    Add manager url to section js

SnowCreative commented:
OK, a small beef here. "Login" is not a verb. It's a noun (and sometimes used as an adjective). "Log in" is correct for use as a verb, as in Christian's latest screen shot above ("Log in to access the manager"). So the button should say "Log In", like it does in the Mailchimp screen shot. If "Login" were a verb, then you would say "Keep me loginned for 2 weeks"!

ghost commented:
@rainbowtiger - give it up. Conference and reference also aren't verbs. One confers with someone or has a conference, and one refers to something or makes a reference, but I haven't seen either of them used properly for decades.

Mark-H commented:
PRs to further improve the login are welcome. :)

SnowCreative commented:
"Reference" has officially been a verb for a while now, and even "conference" is listed as a verb in some dictionaries, although that's more of a stretch. "Login" is not listed as a verb anywhere. Furthermore, when people use "login" as a verb, they pronounce it "log in", with equal stress on both words and a slight separation between them, rather than just accenting the first syllable as they do with "login" as a noun. The misuse of "it's" as a possessive pronoun is also rampant right now, but that doesn't mean we should just accept that and allow its usage. And, I doubt that any dictionaries or grammar texts will ever change to include "it's" as a pronoun rather than a contraction. 

My opinion is that if we want to make something look as professional as possible, using correct grammar helps toward that end.

P.S. Sorry, Mark, I don't know how to do PRs. Maybe one of these days I'll get around to learning how that works.

Mark-H commented:
This PR was already merged so is not the place to discuss it. I doubt anyone would be opposed to improving the English grammar, so please do send a PR (or create a new issue if you're not comfortable with git) for the correction.
